tidb-lightning: update docs of the CSV settings (#6386) (#6815)
pingcap/docs: tidb-lightning/migrate-from-csv-using-tidb-lightning.md

```toml
[mydumper.csv]
# Separator between fields. Must not be empty.
separator = ','
# Quoting delimiter. Empty value means no quoting.
delimiter = '"'

> # Line terminator. Empty value means both "\n" (LF) and "\r\n" (CRLF) are line terminators.
> terminator = ''

Reviewer: ```suggestion
```

trim-last-separator = false
```

> In all string fields such as `separator`, `delimiter` and `terminator`, if the input involves special characters, you can use backslash escape sequence to represent them in a *double-quoted* string (`"…"`). For example, `separator = "\u001f"` means using the ASCII character 0x1F as separator.

Reviewer: ```suggestion
In all string fields such as `separator` and `delimiter`, if the input involves special characters, you can use backslash escape sequence to represent them in a *double-quoted* string (`"…"`). For example, `separator = "\u001f"` means using the ASCII character 0x1F as separator.
```

> Additionally, you can use *single-quoted* strings (`'…'`) to suppress backslash escaping. For example, `terminator = '\n'` means using the two-character string: a backslash followed by the letter "n", as the terminator.

Reviewer: ```suggestion
Additionally, you can use *single-quoted* strings (`'…'`) to suppress backslash escaping. For example, `separator = '\t'` means using the two-character string: a backslash followed by the letter "t", as the separator.
```

[RFC 4180]: https://tools.ietf.org/html/rfc4180

> ### `terminator`
> 
> - Defines the line terminator.
> - If `terminator` is empty, both `"\r"` (U+000D Carriage Return) and `"\n"` (U+000A Line Feed) are used as terminator.
> - Corresponds to the `LINES TERMINATED BY` option in the `LOAD DATA` statement.
> 

Reviewer: ```suggestion
```

- When `trim-last-separator = false`, this is interpreted as a row of 5 fields `('A', '', 'B', '', '')`.
- When `trim-last-separator = true`, this is interpreted as a row of 3 fields `('A', '', 'B')`.

> - This option is deprecated, because the behavior with multiple trailing separators is not intuitive. Use the `terminator` option instead. If your old configuration was
> 
>     ```toml
>     separator = ','
>     trim-last-separator = true
>     ```
> 
>     we recommend changing this to
> 
>     ```toml
>     separator = ','
>     terminator = ",\n"
>     ```
> 

Reviewer: ```suggestion
```

### Non-configurable options

TiDB Lightning does not support every option supported by the `LOAD DATA` statement. Some examples:

> * The line terminator must only be CR (`\r`), LF (`\n`) or CRLF (`\r\n`), which means `LINES TERMINATED BY` is not customizable.

Reviewer: put this back

strict-format = true
```

Currently, a strict CSV file means every field occupies only a single line. In other words, one of the following must be true:

* Delimiter is empty, or

> * Every field does not contain the terminator itself. In the default configuration, this means every field does not contain CR (`\r`) or LF (`\n`).

Reviewer: ```suggestion
* Every field does not contain CR (`\r`) or LF (`\n`).
```

not-null = false
null = '\N'
backslash-escape = true

> trim-last-separator = false

Reviewer: put this back

not-null = false
null = 'NULL'
backslash-escape = false

> trim-last-separator = false

Reviewer: put this back

header = false
not-null = true
backslash-escape = false

> trim-last-separator = false

Reviewer 2: ```suggestion
trim-last-separator = true
```

[mydumper.csv]
separator = '|'
delimiter = ''

> terminator = "|\n"

Reviewer: ```suggestion
```

header = false
not-null = true
backslash-escape = false

> trim-last-separator = true

Reviewer: put this back
